### PR TITLE
Kelsha Darby - User Story 10

### LINKED ISSUE #13: User Story 10, User Registration
As a visitor
When I click on the 'register' link in the nav bar
Then I am on the user registration page ('/register')
And I see a form where I input the following data:
- my name
- my street address
- my city
- my state
- my zip code
- my email address
- my preferred password
- a confirmation field for my password

When I fill in this form completely,
And with a unique email address not already in the system
My details are saved in the database
Then I am logged in as a registered user
I am taken to my profile page ("/profile")
I see a flash message indicating that I am now registered and logged in

### PR DESCRIPTION
**Pull request by kelshadarby**
### The following changes were made

_Database_
- Created `user` table
    - Holds users unique email and encrypted password
- Created `user_details` table
    - Holds all info about the user (name and address)

_Tests_
- Created model tests for the `users` table and the `user_details` table
- Created a user registration feature test

_Models_
- Created a `user` model
     - This model verifies that the email is unique and encrypts the password
- Created a `user_details` model
     - This model verifies that all user info fields are filled in

_Controllers_
- Created the user controller
     - Added`#create` method
         - This method creates a new `user` object and a `user_details` object and flashes a message when it is complete
- Created welcome controller with an empty `#index` method

_View_
- Created welcome view
- Created new user registration form

### CODE REVIEW

**Comment by fredrondina** [comment]
Nice work with the one to one relationship, didn't know about has_one

File: app/controllers/welcome_controller.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class WelcomeController < ApplicationController
+  def index
```
Comment (brian-greeson) [inline app/controllers/welcome_controller.rb:2]: Empty methods do nothing

**Review by brian-greeson — APPROVED** [review]
Great job.

**Comment by brian-greeson** [comment]
Closes: #13

**Pull request merged**